### PR TITLE
[Settings] Add Section tabs

### LINKED ISSUE #52277: [Settings] Add horizontal tabs
Add horizontal tabs for sections. See pfHfG4-84-p2 for Figma files. Populate the tabs with data from https://github.com/woocommerce/woocommerce/issues/52274

Image

### PR DESCRIPTION
**Pull request by psealock**
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

Closes https://github.com/woocommerce/woocommerce/issues/52277

Adds tabs for pages with sections. See LSwxXIhP15A1YvZWnkgihH-fi=350-48825 for designs

image

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Install/Activate Gutenberg
2. Turn on `settings` feature flag
3. WooCommerce > Settings
4. Make sure url, tabs, and content react as you would expect

### Changelog entry

-   [ ] Automatically create a changelog entry from the details below.

-   [x] This Pull Request does not require a changelog entry. (Comment required below)

<details>

<summary>Changelog Entry Details</summary>

#### Significance

-   [x] Patch
-   [ ] Minor
-   [ ] Major

#### Type

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [x] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Message 

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

Adds section tabs to Modern Settings

</details>

### CODE REVIEW

**Review by chihsuan — APPROVED** [review]
Fantastic work, tested well. 👍

File: packages/js/settings-editor/src/components/header/index.tsx [hunk 1]
```diff
@@ -0,0 +1,34 @@
+/**
+ * External dependencies
+ */
+import { createElement } from '@wordpress/element';
+import {
+	// @ts-expect-error missing types.
```
Comment (adrianduffell) [inline packages/js/settings-editor/src/components/header/index.tsx:6]: Wondering, should we contribute these types upstream?

**Review by adrianduffell — APPROVED** [review]
Nice, it tested well for me! Just left a question about the missing types 🚀

File: packages/js/settings-editor/src/components/header/index.tsx (hunk 1 above)
Comment (psealock) [reply packages/js/settings-editor/src/components/header/index.tsx:6]: 👍🏽

**Pull request merged**